jsonschema-maven-plugin/README.md

```diff
@@ -0,0 +1,133 @@
+    </configuration>
+</plugin>
+```
+This will use the default configuration of the generator. This will enable all of the modules that are part of the generator. 
+
+
+### Configuring file locations
+```xml
+<configuration>
+    <className>com.myOrg.myApp.myclass</className>
+    <schemaFileName>mySchema.json</schemaFileName>
+    <schemaFilePath>src/main/resources/schemas</schemaFilePath>
+</configuration>
+```
+- The default name of the file is the name of the class extended with ".schema.json"
+- The default path is "src/main/resources"
+### Configuring schema version
+The version of JSON Schema that is to be used can be configed with the schemaVersion element.
+```xml
+<configuration>
+    <className>com.github.victools.jsonschema.plugin.maven.TestClass</className>
+    <schemaVersion>DRAFT_2019_09</schemaVersion>
+</configuration>
+```
+Allowed values are:
+- DRAFT_2019_09
+- DRAFT_7
+
+The last one is the default
+### Configuring modules
+When you want to have tighter control over the modules that are to be used during generation, the <module> element can be used to define them 
+```xml
+<configuration>
+    <className>com.myOrg.myApp.myclass</className>
+    <modules>
+        <module>
+            <name>Jackson</name>
+        </module>
+    </modules>
+</configuration>  
+```
+This configuration will only enable the Jackson module.
+### Defining the options for a module
+```xml
+<configuration>
+    <className>com.myOrg.myApp.myclass</className>
+    <modules>
+        <module>
+            <name>Jackson</name>
+            <options>
+                <option>FLATTENED_ENUMS_FROM_JSONVALUE</option>
+            </options>
+        </module>
+    </modules>
+</configuration>  
+```
+in this way the options as supported by the module can be specified.
+### Defining the use of custom modules
+To enable a custom module in the generation the following construct can be used:
+```xml
+<configuration>
+    <className>com.myOrg.myApp.myclass</className>
+    <modules>
+        <module>
+            <name>MySchemaModule</name>
+            <className>com.myOrg.myApp.MyAddonModule</className>
+        </module>
+    </modules>
+</configuration>  
+```
+Make sure your custom module is on the classpath.
+Setting options on custom modules is not yet supported.
+### Full example
+```xml
+<build>
+    <plugins>
+        <plugin>
+            <groupId>com.github.victools</groupId>
+            <artifactId>jsonschema-maven-plugin</artifactId>
+            <version>4.8.0</version>
+            <configuration>
+                <className>com.github.victools.jsonschema.plugin.maven.TestClass</className>
+                <schemaVersion>DRAFT_2019_09</schemaVersion>
+                <schemaFileName>testclass.json</schemaFileName>
+                <schemaFilePath>result/schemas</schemaFilePath>
+                <options>
+                    <option>DEFINITIONS_FOR_ALL_OBJECTS</option>
+                    <option>OTHER_OPTION</option>
```

Comment:
Let's pick an existing second option for this example.
And I assume the `OptionPreset.PLAIN_JSON` is being used by default? That should be the most common I guess, or should it be possible to configure that preset as well (to avoid a long list of individual options)?

Since the `OptionPreset` is currently fixed, how would one go about removing an `Option`? E.g. if you don't want the `Option.SCHEMA_VERSION_INDICATOR` to be included?

Reply:
- Changed to NULLABLE_FIELDS_BY_DEFAULT. Is that good one?
- OptionPreset.PLAIN_JSON is indeed used by default. This is currently not configurable.
- The current approach is that the default is OptionPreset.PLAIN_JSON, and all the options from the <options> are added the configBuilder using the with() method. without() is currently not supported.
Let's have a discussion how we can add this feature.

Reply:
Maybe merge what we have now and create an issue to resolve this.

Reply:
- The `Option.NULLABLE_FIELDS_BY_DEFAULT` sounds alright for this example, yes.
- Fair enough to tackle the preset topic separately. Maybe we could also keep it simple here and only have a `boolean` configuration flag for disabling any preset to be applyied (i.e. an empty `new OptionPreset()` instead). That way, full control is possible (although with a long list of individual options) and the default behavior can still be `OptionPreset.PLAIN_JSON`.